### DISCUSSION TITLE
Testing new logic without front

### DISCUSSION THREAD

**Discussion by Disregard-Therest** [serverpod/serverpod#1971] [Q&A] [answered]
Hi,
please, advise what is the best way to develop serverpod functionality without flutter part.

I mean, I have flutter developers in team and I want to prepare backend logic for them to do front-end tasks, while they are not well prepared with serverpod yet.

However I can't actually ensure that functionality is working properly without creating the flutter part which is pretty tedious for me.

**Comment by SandPod** [comment, marked as answer]
Hey!

After project creation, Serverpod does not touch the flutter project provided. You are free to remove it if you don't want it bundled with your backend server code.

Once you would like to start using the generated client in whatever flutter project you have created you can simply import the `serverpod_flutter` and your generated `<you_project_name>_client` package.

When generating the project serverpod only touches the `_client` and the `_server` directories.

That way you can create tests on any level you would like utilizing only the server code or only the client code (without the flutter part). You can also build functionality in the `_client` package that is later used by the flutter app. Just make sure to not modify any automatically generated files.

**Reply by SandPod** [reply]
Some additional info: If you would like to create a `client` you can use to call the endpoints in a test, you can take inspiration from how we run our e2e tests for the framework.

A simple example here where we initialize our generated client and call endpoints on the server: https://github.com/serverpod/serverpod/blob/ca93cf9f7c23017ffe8d1ef893d156886c582e77/tests/serverpod_test_server/test_e2e/database_operations/generated_operations/general_interactions_test.dart#L8

Just note that you need a running server for it to work.
